libraries/api/discussion_helper.cpp

```diff
@@ -196,15 +218,52 @@ namespace golos { namespace api {
             auto vshares = db.calculate_vshares(d.net_rshares.value > 0 ? d.net_rshares.value : 0);
 
             u256 r2 = to256(vshares); //to256(abs_net_rshares);
+            r2 = (r2 * d.reward_weight) / STEEMIT_100_PERCENT;
             r2 *= pot.amount.value;
             r2 /= total_r2;
 
+            uint64_t payout = static_cast<uint64_t>(r2);
+
+            payout = std::min(payout, uint64_t(d.max_accepted_payout.amount.value));
+
+            uint128_t reward_tokens = uint128_t(payout);
+
+            share_type author_tokens = reward_tokens.to_uint64();
+
+            share_type curation_tokens = ((reward_tokens * db.get_curation_rewards_percent())
+                / STEEMIT_100_PERCENT).to_uint64();
+            author_tokens -= curation_tokens;
+            if (d.total_vote_weight > 0 && d.allow_curation_rewards) {
+                auto crs_claim = get_curator_rewards_claim(d, curation_tokens);
+                author_tokens += (curation_tokens - crs_claim);
+                d.pending_curator_payout_gests_value = asset(crs_claim, STEEM_SYMBOL) * props.get_vesting_share_price();
+            }
+
+            uint32_t benefactor_weights = 0;
+            for (auto &b : d.beneficiaries) {
+                benefactor_weights += b.weight;
+            }
+            auto total_beneficiary = (author_tokens * benefactor_weights) / STEEMIT_100_PERCENT;
+            author_tokens -= total_beneficiary;
+            d.pending_benefactor_payout_gests_value = (asset(total_beneficiary, STEEM_SYMBOL) * props.get_vesting_share_price());
+
+            auto sbd_steem = (author_tokens * d.percent_steem_dollars) / (2 * STEEMIT_100_PERCENT);
+            auto vesting_steem = asset(author_tokens - sbd_steem, STEEM_SYMBOL);
+            d.pending_author_payout_gests_value = vesting_steem * props.get_vesting_share_price();
+            auto to_sbd = asset((props.sbd_print_rate * sbd_steem) / STEEMIT_100_PERCENT, STEEM_SYMBOL);
+            auto to_steem = asset(sbd_steem, STEEM_SYMBOL) - to_sbd;
+            d.pending_payout_value = db.to_sbd(to_sbd + to_steem + vesting_steem);
+
+            d.pending_author_payout_golos_value = to_steem;
+            d.pending_author_payout_value = db.to_sbd(to_sbd);
```

Comment:
`pending_author_payout_value` should have value `db.to_sbd(to_sbd + to_steem + vesting_steem);`

Reply:
Where is `pending_author_payout_gbg_value` ?

Reply:
@afalaleev It is `pending_author_payout_value`. If you have some problems with understanding such things in changes, you should reference to HTML-JS in checklist.

Reply:
Before changes there were:
- `pending_payout_value` - full post payout (author+curator+benefactor) converted to GBG 
- `total_pending_payout_value` - full post payout ((author+curator+benefactor)+ all comments payout (author+curator+benefactor) converted to GBG

We can't change or remove these fields, because users have scripts with calculation around them values.

Whats want to add:
- `pending_payout_value` (should still have the full payout value converted to GBG).
1. `pending_author_payout_value` - author payout value converted to GBG
1. `pending_curator_payout_value` - curator payout value converted to GBG
1. `pending_benefactor_payout_value` - benefactor payout value converted to GBG
- `total_pending_payout_value` - total payout value converted to GBG

Ok. Now
- curators get their payout in GESTS, that is why we add `pending_curator_payout_gests_value`
- beneficiaries get their payout in GESTS, that is why we add `pending_benefactor_payout_gests_value`
- authors get their payout in 3 parts:
1. GESTS - `pending_author_payout_gests_value`
1. GBG - `pending_author_payout_gbg_value`
1. GOLOS - `pending_author_payout_golos_value`

Users shouldn't try to understand why:
- one version of golosd returns part of payout in GBG in `pending_author_payout_value`, other version of golosd returns full post payout converted to GBG in `pending_author_payout_value`
- `pending_author_payout_value` doesn't contain full author payout converted in GBG, and contains only part of payout paid in GBG